tokio/src/macros/test.rs

```diff
@@ -0,0 +1,25 @@
+/// Helper for writing a fallible unix-specific doctests.
+#[macro_export]
+macro_rules! doctest_unix {
+    ($($body:tt)*) => {
+        #[cfg(unix)]
+        #[tokio::main]
+        async fn main() -> Result<(), Box<dyn ::std::error::Error>> { $($body)* }
+
+        #[cfg(not(unix))]
+        fn main() {}
+    }
+}
```

Comment:
Is this public? If so, can we gate it behind the `docsrs` thing?

Reply:
It *is* public, but it was supposed to be `#[doc(hidden)]` (will fix).

I'll mention gating behind `--cfg docsrs` briefly here. The unfortunate side effect of doing that would be that running *any* tests would require specifying `docsrs`. Cause the macro is used in all platform-specific doctests.

Reply:
Ah. I see what you mean now from looking at what you did in #3770. *Hide* the module completely unless `docsrs` is specified or it's the correct platform. If `docsrs` is enabled, do the stubbing.

This fixes doc testing, because it doesn't enable `doc`.

I've incorporated that approach here now. So the macros are no longer needed.